e2e_tests/tests/deploy/test_local.py

```diff
@@ -114,6 +114,24 @@ def test_cluster_down() -> None:
     assert len(containers) == 0
 
 
+@pytest.mark.det_deploy_local
+def test_ee_cluster_up() -> None:
+    name = "test_ee_cluster_up"
+
+    with resource_manager(Resource.CLUSTER, name, {}, ["no-gpu", "enterprise-edition"]):
+        container_name = name + "_determined-master_1"
+        client = docker.from_env()
+
+        containers = client.containers.list(filters={"name": container_name})
+        assert len(containers) > 0
+
+        for container in containers:
+            if "Config" in container.attrs and "Image" in container.attrs["Config"]:
+                assert "hpe-mlde-master" in container.attrs["Config"]["Image"]
```

Comment:
I like the direction you went. Validating the correct Docker container was used seems closer to the behavior under test than checking whether the master in the container is EE-enabled.